### ISSUE TITLE
Check if team has enough point for hint request submission

### ISSUE THREAD

**Issue by KoleckoOrg** [mfrodl/kolecko#43]
Dunno what happens ATM as points are not printed anywhere, but my guess is that the points just go into negative.

This should not be possible.

**Comment by KoleckoOrg** [comment]
Well in theory this should not be possible but I somehow managed to get to -480 points. Unfortunately I can't reproduce now...

**Comment by KoleckoOrg** [comment]
There are two things broken as far as I can see.
1- balance checking on help request submission does not work
2- points are not substracted on help request submission
I'll prepare a pull request in a moment.

Unfortunately, this does not explain how I ended up with a negative points so there is at least one more bug somewhere...

**Comment by KoleckoOrg** [comment]
I saw you reverted the balance checking change 0ff5370a6cdd6ae76157afe42ca5e983c6915481, would you have time to explain what is going on there?

**Comment by mfrodl** [comment]
See my comment on the pull request: https://github.com/mfrodl/kolecko/pull/46#pullrequestreview-283424223

**Comment by mfrodl** [comment]
Anyway, I have a suspicion that the buggy behaviour has been fixed by https://github.com/mfrodl/kolecko/commit/875cf39dd63d1d2034f7b6a0586b0b7d4675b47f.

`@hint_request.team` and `current_team` are two distinct in-memory representations of the same database object from the `teams` table. They can be modified independently of each other and the changes do not take effect in the database until a `save` method is called. Previously, we updated `current_team.points` but the validator checks the value of `@hint_request.team.points`, hence the inconsistency.

At least I guess so.
